File: sinaleasy-front/src/domain/City.ts

import { Signal } from "./Signal";

export class City{
    cityId: number | undefined;
    name: string | undefined;
    state: string | undefined;
    rating: number | undefined;
    signals: Array<Signal> | undefined;

    constructor(cityId?: number, name?: string, state?: string, rating?: number, signals?: Array<Signal>){
        this.cityId = cityId;
        this.name = name;
        this.state = state;
        this.rating = rating;
        this.signals = signals;
    }
}

File: sinaleasy-back/src/main/java/com/sinalez/sinaleasy_back/entities/City.java
package com.sinalez.sinaleasy_back.entities;

import java.io.Serializable;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonBackReference;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "TB_Cities")
@Getter
@Setter
public class City implements Serializable {
    @Id
    private String cityId;

    @NotBlank private String name;
    @NotBlank private String state;
    @NotNull private Integer rating;

    @JsonBackReference
    @OneToMany(mappedBy = "city", cascade = CascadeType.ALL, orphanRemoval = true) 
    private List<Signal> signals;
}
